hypothesis/client
# Convert StreamController to a component

**This is similar to https://github.com/hypothesis/client/pull/313 but for the stream instead of the sidebar**

Convert the `StreamController` controller which provides the logic for
the content area of the stream into a `<stream-content>` component.

As part of this I converted the code from CoffeeScript to JS and removed anything unused. The controller and test code is mostly a line-by-line translation.

----

**Testing**

- Load up http://localhost:5000/stream , check that annotation threads load.
- Try a few search queries, check that they load the same results as before.
- If you have enough annotations locally, scroll to the bottom of the view and check that the next batch is loaded
